Source/WebCore/platform/graphics/cg/GraphicsContextCG.cpp

```diff
@@ -1555,6 +1567,15 @@ bool GraphicsContextCG::canUseShadowBlur() const
     return (renderingMode() == RenderingMode::Unaccelerated) && hasBlurredShadow() && !m_state.shadowsIgnoreTransforms();
 }
 
+bool GraphicsContextCG::commitDraws()
```

Comment:
Is this maybe `didFlushDrawing`? Since it doesn't actually commit anything.

Reply:
No, this is completely inverse of "did flush drawing".  Method named "didFlushDrawing" would be called as signal, "I did flush the drawing, do what you need to do". This is totally inverse of what is intended here.

In case my English is poor and you meant "didFlushDrawing" in the meaning of "drawingWasFlushed", it is still the same: the intention and the code as it reads is totally inverse this.

This is the signal for the caller to decide if they need to flush the drawing.

TLDR; the commitDraw() commits the state "did we draw" (as well as potential buffered draws).

The caller here is interested in chunks of time which they decide based on some arbitrary criteria.

So the sequence they query is like:
* commitDraws()
* commitDraws()
* commitDraws()
( This point is to highlight that that is the only interface this part of the caller uses -- it divides the time with these calls and cannot access anything else).

Those commits say to the drawing context :
- please commit all the draws that happened since last time I called
- please tell me if there was a draw that was committed

Whether or not the drawing context *does* any work *today* for this trigger is encapsulated inside the implementation, it is not concern of the interface.

If the implementation was such that it would optimise the draws that negate some previous draws, it might build a buffer of draws. This is the place such an implementation would need to commit the buffer instead of build up more for that purpose. 

Continuing: all buffers have a max length.  So in above scenario, of course the work would be sent further either when the buffer would fill up or when the commit needs it. So obviously in above hypothetical implementation the work would be sent forward not only at commit, but during draws.

The flag represents the state of above mentioned buffer, but with the distinction that the buffer is actually zero length -- e.g. it does not have capacity to defer any of the mentioned draws, but it still can track whether there was a draw or not.

Reply:
I'm also confused by the naming here. It sounds like `commitDraws()` would actually do work, but it just resets the flag. This is really just `std::exchange(m_hasDrawn, false)`. Maybe the call sites should just use that?